packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+# set SBINDIR to /sbin if it is not set
+if [ -z "$SBINDIR" ]; then
+    SBINDIR=/usr/sbin
+fi
+
+# check if system is registered
+subscription-manager identity > /dev/null
+if [ $? -ne 0 ]; then
+    echo "Please register with subscription-manager first, then re-run `$SBINDIR/katello-pull-provider-migrate`."
+    exit 1
+fi
+
+echo "Getting configuration from subscription-manager..."
+KPPTEMPFILE=$(mktemp kpp_tempfile_XXXXXXXX)
+subscription-manager config --list > $KPPTEMPFILE
+CONSUMER_CERT_DIR=$(grep 'consumercertdir' $KPPTEMPFILE | cut -d= -f2 | xargs | sed 's/[][]//g')
+CERT_FILE=$CONSUMER_CERT_DIR/cert.pem
+KEY_FILE=$CONSUMER_CERT_DIR/key.pem
+CA_FILE=$(grep 'repo_ca_cert = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+SERVER_NAME=$(grep 'hostname = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+
+# fail if no server name, cert dir, or ca file
+if [ -z "$SERVER_NAME" ] || [ -z "$CONSUMER_CERT_DIR" ] || [ -z "$CA_FILE" ]; then
+    echo "Unable to determine config from `subscription-manager config --list`; exiting"
+    exit 1
+fi
+
+# fail if client is not registered to a Katello
+if ! grep -q 'prefix = \/rhsm' $KPPTEMPFILE; then
+    echo "Client is not registered to Katello; exiting"
```

Comment:
You may consider not using `Katello` here to reduce branding scope.

Reply:
But if I delete `Katello` it won't be clear that it will exit if the client is registered to RHSM but not to a Katello / Satellite. :thinking:

Reply:
What about
```
Client is registered to RHSM; exiting
```

Reply:
I could see how RHSM is confusing as well. So I might opt for 'Foreman' that at least is the application that is running.

Is checking the configuration of `rhsm.conf` an actual teller of registration though? If I configure `rhsm.conf` to point at a Foreman or smart-proxy, I am not actually registered at that point. Could we use subscription-manager to report registration?

e.g, if not registered I see:

```
# subscription-manager status
+-------------------------------------------+
   System Status Details
+-------------------------------------------+
Overall Status: Unknown

System Purpose Status: Unknown
```